### PR TITLE
finished assignment

### PR DESCRIPTION
**Pull request by michael-duren**

### CODE REVIEW

File: assignment/scripts/3-music-collection.js [hunk 1]
```diff
@@ -1 +1,145 @@
-console.log('***** Music Collection *****')
+console.log('***** Music Collection *****');
+
+const collection = [];
+
+const addToCollection = (title, artist, yearPublished, tracks) => {
+  const album = {
```
Comment (delventhalz) [inline assignment/scripts/3-music-collection.js:6]: Good variable naming.

File: assignment/scripts/3-music-collection.js [hunk 2]
```diff
@@ -1 +1,145 @@
...
+    yearPublished,
+    tracks,
+  };
+  collection.push(album);
+  return album;
+};
+
+const showCollection = (collectionArray) => {
+  console.log(collectionArray.length);
+  collectionArray.forEach((album) => {
```
Comment (delventhalz) [inline assignment/scripts/3-music-collection.js:18]: Oh `.forEach`. What purpose do you serve now that for...of exists? I don't know, but some style guides disagree with me for some reason.

File: assignment/scripts/3-music-collection.js [hunk 3]
```diff
@@ -1 +1,145 @@
...
+    console.log(
+      `\n${album.title} by ${album.artist}, published in ${album.yearPublished}`
+    );
+    console.log('*'.repeat(60));
```
Comment (delventhalz) [inline assignment/scripts/3-music-collection.js:22]: Nice

File: assignment/scripts/3-music-collection.js [hunk 4]
```diff
@@ -1 +1,145 @@
...
+  1990,
+  fearOfABlackPlanetTracks
+);
+
+// test showCollection
+showCollection(collection);
+
+// findByArtist
+const findByArtist = (artist) => {
+  return collection.filter(
```
Comment (delventhalz) [inline assignment/scripts/3-music-collection.js:87]: Nice use of filter.

File: assignment/scripts/3-music-collection.js [hunk 5]
```diff
@@ -1 +1,145 @@
...
+const biggieSmalls = findByArtist('The Notorious B.I.G.');
+console.log('\nSongs by The Notorious B.I.G.', biggieSmalls);
+
+// search
+const search = (searchObj = { artist: '', year: '', trackName: '' }) => {
+  const { artist, year, trackName } = searchObj;
+  // search track, ignoring artist and year
+  if (trackName) {
+    return collection.filter((album) =>
+      album.tracks.find((track) => track.name === trackName)
```
Comment (delventhalz) [inline assignment/scripts/3-music-collection.js:109]: Solid. `some` would also have worked and been a _little_ more semantically meaningful, but great use of array methods here.

**Review by delventhalz — COMMENTED** [review]
Great work. That strong understanding of higher-order functions and array methods is going to serve you. Only problem is now 1979 is stuck in my head.

File: assignment/scripts/3-music-collection.js (hunk 5 above)
Comment (michael-duren) [reply assignment/scripts/3-music-collection.js:109]: I just found out about that one after doing the primelib, I definitely will use in the future!

**Pull request closed without merging**